### PR TITLE
davix: a new package

### PR DESCRIPTION
**Pull request by linxon**
Maintainer: Yury Martynov / @linxon (email@linxon.ru)
Compile tested: 18.06 / 19.07  / [last commit](https://github.com/openwrt/openwrt/commit/1e3bfbafd37ccb32d0ed6618f4886e1dec6643d2)
Run tested: Broadcom BCM63xx

Description:
[Davix](https://github.com/cern-fts/davix) aims to make the task of managing files over HTTP-based protocols simple. It is being developed by IT-ST at CERN, and while the project's purpose is its use on the CERN grid, the functionality offered is generic.

Homepage: https://davix.web.cern.ch https://github.com/cern-fts/davix

### CODE REVIEW

File: net/davix/Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+define Package/davix
+  SECTION:=net
+  CATEGORY:=Network
+  DEPENDS:=+libxml2 +libuuid +libopenssl
+  TITLE:=High-performance file management over WebDAV/HTTP
+  URL:=https://dmc.web.cern.ch/projects/davix
+endef
+
+TARGET_LDFLAGS+= \
+	-static-libstdc++
```
Comment (neheb) [inline net/davix/Makefile]: Why?

File: net/davix/Makefile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+CMAKE_OPTIONS += \
+	-DDAVIX_TESTS=OFF \
+	-DBENCH_TESTS=OFF \
+	-DENABLE_TOOLS=ON \
+	-DENABLE_TCP_NODELAY=ON
+
+CMAKE_OPTIONS += $(if $(CONFIG_IPV6),-DENABLE_IPV6=ON,-DENABLE_IPV6=OFF)
```
Comment (neheb) [inline net/davix/Makefile]: Could just make this part of the original block.

File: net/davix/Makefile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+define Build/InstallDev
```
Comment (neheb) [inline net/davix/Makefile]: Not needed if you use CMAKE_INSTALL.

File: net/davix/Makefile [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/cern-fts/davix/releases/download/R_0_7_5/
+PKG_HASH:=d920ca976846875d83af4dc50c99280bb3741fcf8351d5733453e70fa5fe6fc8
+
+PKG_BUILD_PARALLEL:=1
+
+PKG_MAINTAINER:=Yury Martynov <email@linxon.ru>
+PKG_LICENSE:=LGPL-2.1
```
Comment (neheb) [inline net/davix/Makefile]: should be LGPL-2.0-or-later.

**Comment by diizzyy** [comment]
Ping @linxon

**Pull request closed without merging**